src/index.js

```diff
@@ -18,9 +18,20 @@ import 'babel-polyfill';
 import fs from 'fs';
 import path from 'path';
 import vorpal from 'vorpal';
-import { version } from '../package.json';
+import semver from 'semver';
+import { version, engines } from '../package.json';
 import config from './utils/config';
 
+const nodeVersionError = (required, actual) =>
+	chalk.red(`
+Error: Wanted node version ${required} but has ${actual}
+`);
+
+if (!semver.satisfies(process.version, engines.node)) {
+	console.error(nodeVersionError(engines.node, process.version));
+	process.exit(0);
+}
+
```

Comment:
This logic would probably make more sense in `bin/lisky` since functionality could break there before this file is even reached.